### ISSUE TITLE
What is replacing com.oem.autotest with \0om.oem.autotest in /system/vendor/lib64/hw/com.qti.chi.override.so for?

### ISSUE THREAD

**Issue by shamefulCake1** [sean6541/op9_series_camera_unlocker#2]
I ran strings on my `com.qti.chi.override.so` and com.oem.autotest is nowhere to be found.

What is it for?

**Comment by sean6541** [comment]
com.oem.autotest is the package id of a internal app OnePlus uses for testing. It gets access to a bunch of hidden camera API's. When an app accesses the camera, `strncmp` is called to compare the package id to the string `com.oem.autotest`. `strncmp` only compares up until it reaches a null character, so by replacing the first character of the string `com.oem.autotest` with null, all apps get access to these hidden API's.

As for why it can't be found, what device is your file from?

**Comment by shamefulCake1** [comment]
Ah, so you're saying that com.oem.autotest is the super-priviledged package that we all now imitate?

>As for why it can't be found, what device is your file from?

OnePlus Ace 2 Pro

I guess it how has some different options for restricting camera access.

**Comment by sean6541** [comment]
>Ah, so you're saying that com.oem.autotest is the super-priviledged package that we all now imitate?

Yes, exactly. Unfortunately, full resolution is restricted differently depending on the device and chipset.
